syntax = "proto3";

package creative_campaign;

// BrandComposeRequest is sent when an image needs brand elements added
message BrandComposeRequest {
  string campaign_id = 1;
  string locale = 2;
  string product_id = 3;
  
  // Image to brand
  string image_url = 4;
  string s3_uri = 5;
  
  // Brand configuration
  string logo_s3_uri = 6;
  string primary_color = 7;
  string logo_position = 8;  // top_left, top_right, bottom_left, bottom_right
  
  // Metadata
  string correlation_id = 10;
  string timestamp = 11;
}

// BrandComposed is published when brand elements have been added
message BrandComposed {
  string campaign_id = 1;
  string locale = 2;
  string product_id = 3;
  
  // Branded image details
  string branded_image_url = 4;
  string branded_s3_uri = 5;
  
  // Original image reference
  string original_image_url = 6;
  string original_s3_uri = 7;
  
  // Status
  string status = 8;  // "composed" or "failed"
  string error_message = 9;  // Only set if status is "failed"
  
  // Metadata
  string correlation_id = 10;
  string composed_at = 11;
}
